.github/workflows/sync-repos.yml

```diff
@@ -0,0 +1,87 @@
+          fetch-depth: 0
+      - name: Check if examples folder has changes
+        id: check_example_projects_changes
+        run: |
+          CHANGED_FILES=$(git diff --name-only ${{ github.event.before }} ${{ github.event.after }} -- 'examples/**/*')
+          if [ -n "$CHANGED_FILES" ]; then
+            echo "::set-output name=has_changes::true"
+          else
+            echo "::set-output name=has_changes::false"
+          fi
+      - name: Check for changes in specified path
+        if: steps.check_example_projects_changes.outputs.has_changes == 'true'
+        id: check_changes
+        run: |
+          changed_paths=""
+          for path in examples/*/; do
+            CHANGED_FILES=$(git diff --name-only ${{ github.event.before }} ${{ github.event.after }} -- "${path}*")
+            if [ -n "$CHANGED_FILES" ]; then
+              changed_paths+="$path;"
+            fi
+          done
+          echo "::set-output name=changed_paths::$changed_paths"
+      - name: Update listening repos, create PR, and clean up
+        if: steps.check_changes.outputs.changed_paths != ''
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          # This separates the strings with ; to turn it into an array
+          IFS=';' read -ra changed_paths_array <<< "${{ steps.check_changes.outputs.changed_paths }}"
+          echo "Changed paths array: ${changed_paths_array[@]}"
+
+          # Configure Git
+          git config --global user.name "github-actions[bot]"
+          git config --global user.email "github-actions[bot]@users.noreply.github.com"
+          
+          for path in "${changed_paths_array[@]}"; do
+            # Store the current working directory
+            initial_working_directory=$(pwd)
+
+            # Extract the folder name from the path
+            folder_name=$(basename $path)
+
+            # Check if the listening repo exists
+            repo="https://$GITHUB_TOKEN:@github.com/medplum/${folder_name}.git"
+            REPO_STATUS=$(curl -s -o /dev/null -I -w "%{http_code}" -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" "https://api.github.com/repos/medplum/${folder_name}")
+
+            # If the repo does not exist, create a new repo
+            if [ $REPO_STATUS -eq 404 ]; then
+              CREATE_REPO_PAYLOAD="{\"name\": \"${folder_name}\", \"default_branch\": \"main\"}"
+              CREATE_REPO_RESPONSE=$(curl -s -X POST -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" -H "Content-Type: application/json" -H "Accept: application/vnd.github+json" --data "$CREATE_REPO_PAYLOAD" "https://api.github.com/orgs/medplum/repos")
+              echo "Created new repo: $(echo $CREATE_REPO_RESPONSE | jq -r '.html_url')"
+            fi
+
+            # Clone the corresponding listening repo
+            git clone "$repo"
+
+            # Copy changed files to the listening repo
+            rsync -a --delete "${initial_working_directory}/${path%}"/ ${folder_name}/
```

Comment:
I believe for the safe side we can exclude .git/ path so that the listening repo will have it. And yes, the example directory projects will be fully contained. The expectation is that all of the listening repos will be read only.